request_test.go

```diff
@@ -28,6 +29,41 @@ type AuthError struct {
 	ID, Message string
 }
 
+func TestRequestRace(t *testing.T) {
+	// Issue #225
+
+	ts := createGetServer(t)
+	defer ts.Close()
+
+	req := R()
```

Comment:
@moorereason I have looked at this PR and tried out locally. This is incorrect usage because this test case sharing `Request` object across goroutine. Resty does not work that way. Each request processed in the dedicated request object. 

Usage is -
```go
req := R()
resp, err := req.Get(ts.URL + "/")
```

And #225 issue involves the Request body and user reported data race in request body writing.

Reply:
I agree with you.  I mistakenly thought this was related to #225 when I submitted it (in haste).  I only found this PR while trying to create data races with resty.  I don't know that anyone would actually do this in production if they understand Go concurrency.

My only mild suggestion would be to clarify the docs to say that the "goroutine concurrent safe" guarantee applies on to the resty Client.

Closing this PR.